localstack/aws/protocol/parser.py

```diff
@@ -980,7 +980,9 @@ def parse(self, request: HttpRequest) -> Tuple[OperationModel, Any]:
         """Handle virtual-host-addressing for S3."""
         if (
             # TODO implement a more sophisticated determination if the host contains S3 virtual host addressing
-            not request.host.startswith("s3.")
+            not request.host == "localhost"
```

Comment:
When working on the ASF service router, I stumbled upon the function used for the service routing which checks if a host header indicates the S3 virtual host addressing: 
https://github.com/localstack/localstack/blob/d12d037398ec8c1db26be15ac60e481579397e36/localstack/services/s3/s3_utils.py#L90-L99

Maybe we could directly use this function here?